File: src/scandoc/tui/screens/pipeline.py
"""
Pipeline Configuration Screen renderer for scanDOC TUI.
"""

from rich.panel import Panel
from rich.table import Table
from rich.text import Text

from scandoc.tui.state import TuiState


def render_pipeline_screen(state: TuiState, selected_idx: int = 0) -> Panel:
    """Render interactive pipeline editor screen."""
    cfg = state.pipeline_config

    stages = [
        ("1", "Native PDF Fast-Path", True, "PyPdfium2"),
        ("2", "OCR Ingestion Engine", True, "RapidOCR"),
        ("3", "DocLayNet Layout Detection", True, "RT-DETR"),
        ("4", "Reading Order Analysis", True, "XY-Cut"),
        ("5", "Table Structure Recognition", True, "SLANet"),
        ("6", "Formula Vision Engine", True, "LaTeX-OCR"),
        ("7", "Multimodal VLM Analysis", cfg.enable_vlm_fallback, "SmolVLM"),
        ("8", "Agentic Routing Strategy", True, cfg.routing_mode.upper()),
        ("9", "Offline Mode Toggle", state.is_offline(), "ENABLED" if state.is_offline() else "DISABLED"),
    ]

    table = Table(show_header=True, box=None, padding=(0, 1))
    table.add_column("Key", style="bold cyan", width=4)
    table.add_column("Status", width=8)
    table.add_column("Pipeline Stage", style="bold white")
    table.add_column("Active Provider", style="dim green")

    for i, (key, label, enabled, provider) in enumerate(stages):
        status_str = "[✓] ON" if enabled else "[○] OFF"
        status_style = "bold green" if enabled else "dim red"
        row_style = "bold yellow reverse" if i == selected_idx else "white"

        table.add_row(f"[{key}]", Text(status_str, style=status_style), f"{'› ' if i == selected_idx else '  '}{label}", provider, style=row_style)

    subtitle = "Space/Enter: Toggle Stage | 9: Offline Mode | Esc: Back | [P]: Run Pipeline"

    return Panel(
        table,
        title="[bold blue]Pipeline Configuration Editor[/bold blue]",
        subtitle=subtitle,
        border_style="magenta",
        padding=(1, 2),
    )

File: internal/tui/screens/pipeline/pipeline.go
package pipeline

import (
	"fmt"
	"strings"

	"scandoc/internal/tui/state"
	"scandoc/internal/tui/styles"
)

func Render(st *state.AppState, selectedIdx int) string {
	var b strings.Builder
	b.WriteString(styles.HeaderStyle.Render(" Pipeline Stage Configuration & Routing ") + "\n\n")

	b.WriteString(styles.ActiveItemStyle.Render(fmt.Sprintf("Current Agentic Routing Mode: [%s]", strings.ToUpper(st.PipelineConfig.RoutingMode))) + "\n\n")

	stages := []struct {
		Label   string
		Enabled bool
		Desc    string
	}{
		{"Optical Character Recognition (OCR Engine)", st.PipelineConfig.EnableOCR, "RapidOCR PP-OCRv4 ONNX"},
		{"Visual Layout Analysis (Layout Detector)", st.PipelineConfig.EnableLayout, "RT-DETR DocLayNet ONNX"},
		{"Neural Table Structure Recognition", st.PipelineConfig.EnableTable, "SLANet Table ONNX"},
		{"Mathematical Formula LaTeX Parsing", st.PipelineConfig.EnableFormula, "Pix2Text LaTeX-OCR ONNX"},
	}

	for i, stage := range stages {
		check := "[ ]"
		if stage.Enabled {
			check = "[✓]"
		}

		line := fmt.Sprintf("%s %-45s | %s", check, stage.Label, stage.Desc)
		if i == selectedIdx {
			b.WriteString(styles.ActiveItemStyle.Render("› "+line) + "\n")
		} else {
			b.WriteString(styles.NormalItemStyle.Render("  "+line) + "\n")
		}
	}

	b.WriteString("\n" + styles.FooterStyle.Render("Space: toggle stage | r: cycle routing mode (adaptive/fast/deep/fallback) | Esc: Home"))
	return styles.PanelStyle.Render(b.String())
}
